tune-tikv-thread-performance.md

```diff
@@ -0,0 +1,86 @@
+
+    - If the deployed machine has a very small number (less than or equal to 8) of CPU cores, you can consider setting the `server.grpc-concurrency` configuration item to `2`.
+    - If the deployed machine has very high configuration, TiKV undertakes a large number of read and write requests, and the value of gRPC poll CPU that monitors Thread CPU on Grafana exceeds 80% of `server.grpc-concurrency`, then you can consider appropriately increasing the value of `server.grpc-concurrency` to keep the thread pool usage rate below 80% (that is, the metric on Grafana is lower than the `80% * server.grpc-concurrency` value).
+
+* The Scheduler thread pool.
+
+    When TiKV detects that the number of machine CPU cores is larger than or equal to 16, the default size configuration (`storage.scheduler-worker-pool-size`) of the Scheduler thread pool is `8`; when TiKV detects that the number of machine CPU cores is smaller than 16, the default configuration is `4`.
+
+    This thread pool is mainly used to transform complex transaction requests into simple key-value read and write requests. However, **the scheduler thread pool itself does not perform any write operation**.
+
+    - If it detects a transaction conflict, then this thread pool returns the conflict result to the client in advance.
+    - If no conflict is detected, then this thread pool merges the key-value requests (that needs to be written) into a Raft log and hands it over to the Raftstore thread for Raft log replication.
+
+    Generally speaking, to avoid excessive thread switching, it is best to ensure that the utilization rate of the Scheduler thread pool is between 50% and 75%. If the thread pool size is `8`, then it is recommended to keep `TiKV-Details.Thread CPU.scheduler worker CPU` on Grafana between 400% and 600%.
+
+* The Raftstore thread pool.
+
+    The Raftstore thread pool is the most complex thread pool in TiKV. The default size (`raftstore.store-pool-size`) is `2`. All write requests are written into RocksDB in the way of `fsync` from the Raftstore thread, unless you manually set `raftstore .sync-log` to `false`. Setting `raftstore.sync-log` to `false` can improve some write performance, but also increases the risk of data loss in the case of machine failure).
+
+    Due to the presence of I/O, Raftstore threads cannot reach 100% CPU theoretically. To reduce the number of disk writes as much as possible, you can put together multiple write requests and write them to RocksDB. It is best to control the overall CPU usage below 60% (Because the default number of threads is `2`, it is recommended to keep `TiKV-Details.Thread CPU.Raft store CPU` on Grafana within 120%). Do not increase the size of the Raftstore thread pool to improve write performance without thinking, because this might increase the disk burden and degrade performance.
```

Comment:
```suggestion
    Due to I/O, Raftstore threads cannot reach 100% CPU usage theoretically. To reduce disk writes as much as possible, you can put together multiple write requests and write them to RocksDB. It is recommended to keep the overall CPU usage below 60% (If the default number of threads is `2`, it is recommended to keep `TiKV-Details.Thread CPU.Raft store CPU` on Grafana within 120%). Do not increase the size of the Raftstore thread pool to improve write performance without thinking, because this might increase the disk burden and degrade performance.
```